lib/eth/EthRpc.js

```diff
@@ -147,7 +147,13 @@ class EthRPC {
         await new Promise(async (resolve, reject) => {
           try {
             let confirmations = 0;
+            let timeout = new Date().getTime() + 45000;
```

Comment:
It would not be abnormal for a tx to take 45 sec to be mined. How about 120 sec? (10 avg block times)